File: src/main/java/com/example/backend/dto/tarot/TarotCardDto.java
package com.example.backend.dto.tarot;

import java.util.UUID;

public record TarotCardDto(UUID id,
                           TarotCardTypeDto cardType,
                           String name,
                           String description,
                           String reversedDescription,
                           String advice,
                           boolean reversed,
                           UUID imageId) {
    public TarotCardDto {
    }

    public TarotCardDto(String name) {
        this(null, null, name, null, null, null, false, null);
    }
}

File: src/main/kotlin/com/example/backend/dto/tarot/TarotCardDto.kt
package com.example.backend.dto.tarot

import java.util.*

data class TarotCardDto(
    var id: UUID? = null,
    var cardType: TarotCardType? = null,
    var name: String? = null,
    var description: String? = null,
    var reversedDescription: String? = null,
    var advice: String? = null,
    var reversed: Boolean? = false,
    var imageId: UUID? = null
)
